rasa/shared/nlu/training_data/features.py

```diff
@@ -43,6 +65,207 @@ def is_dense(self) -> bool:
         """
         return not self.is_sparse()
 
+    @staticmethod
+    def filter(
+        features_list: List["Features"],
+        attributes: Optional[Iterable[Text]] = None,
+        type: Optional[Text] = None,
+        origin: Optional[List[Text]] = None,
+        is_sparse: Optional[bool] = None,
+    ) -> List["Features"]:
+        """Filters the given list of features.
+
+        Args:
+          features_list: list of features to be filtered
+          attributes: List of attributes that we're interested in. Set this to `None`
+            to disable this filter.
+          type: The type of feature we're interested in. Set this to `None`
+            to disable this filter.
+          origin: If specified, this method will check that the exact order of origins
+            matches the given list of origins. The reason for this is that if
+            multiple origins are listed for a Feature, this means that this feature
+            has been created by concatenating Features from the listed origins in
+            that particular order.
+          is_sparse: Defines whether all features that we're interested in should be
+            sparse. Set this to `None` to disable this filter.
+
+        Returns:
+            sub-list of features with the desired properties
+        """
+        filtered = features_list
+        if attributes is not None:
+            attributes = set(attributes)
+            filtered = [f for f in filtered if f.attribute in attributes]
```

Comment:
I don't see any guidelines on that and we use single letter variables (inside listcomps) in other functions too? But I can change it, no probem. -  Btw we should someday reconsider the naming conventions with respect to "features" (because we have "features" vs "features.features")

Reply:
Had a look again and it's making lots of expressions less readable (because it's more lengthy and because the variable name itself can still not clarify whether it describes a `Features` or `np.ndarray/scipy.sparse.spmatrix`)